tmt/utils/__init__.py

```diff
@@ -4358,6 +4358,30 @@ def increment(
 
                 raise GeneralError(message) from error
 
+        # Handle GitHub-specific responses
+        # https://docs.github.com/en/rest/using-the-rest-api/rate-limits-for-the-rest-api?apiVersion=2022-11-28#exceeding-the-rate-limit
+        response = cast(requests.Response, kwargs.get('response'))
+        if response is not None and 'X-GitHub-Request-Id' in response.headers:
+            headers = response.headers
+
+            # Handle rate limiting
+            if response.status_code in (403, 429):
+                # Primary rate limit exceeded
+                if ('X-RateLimit-Remaining' in headers and
+                        int(headers['X-RateLimit-Remaining']) == 0):
+                    reset_time = int(headers['X-RateLimit-Reset'])
+                    # Wait until the rate limit resets
+                    wait_time = reset_time - int(time.time())
+                    if wait_time > 0:
+                        time.sleep(wait_time)
+                        return super().increment(*args, **kwargs)
+
+                # Secondary rate limit - respect Retry-After if provided
+                if 'Retry-After' in headers:
+                    retry_after = int(headers['Retry-After'])
+                    time.sleep(retry_after)
+                    return super().increment(*args, **kwargs)
```

Comment:
this line does not seem to be needed, 4385 will be executed next?

Reply:
done